Version 2.0: Major updates for 2026 season

dpgarceau opened the pull request:
## Core Features
- Per-type round numbering (KNOWN, UNKNOWN, FREESTYLE independent counters)
- Incomplete KNOWN round detection and warning
- FREESTYLE support with full scoring functionality
- Sequence validation with warnings shown and graceful fallbacks
- Score mismatch resolution tools for scorekeepers
- Retrieve unknown figures package from Score (4.71 or greater)

## Admin Pages Redesign
- /admin/comp - Competition settings management
- /admin/device - Device/system management with update capability
- /admin/scores - Score management tools

## UI Improvements
- Admin confirmation modal before accessing admin pages
- Battery indicator improvements (warnings only)
- Improved pilot selection and judging page layouts
- Disabled buttons for unavailable sequence types
- Improved error handling

## Technical
- Configurable timeouts and settings API
- Updated audio files including freestyle caller and new user help

dpgarceau committed:
Version 2.0: Major updates for 2026 season

## Core Features
- Per-type round numbering (KNOWN, UNKNOWN, FREESTYLE independent counters)
- FREESTYLE support with full scoring functionality
- Sequence validation with graceful fallbacks
- Score mismatch resolution tools for scorekeepers

## Admin Pages
- /admin/comp - Competition settings management
- /admin/device - Device/system management with update capability
- /admin/scores - Score management tools

## UI Improvements
- Admin confirmation modal before accessing admin pages
- Battery indicator improvements (warnings only)
- Improved pilot selection and judging page layouts
- Disabled buttons for unavailable sequence types

## Technical
- Configurable timeouts and settings API
- Multipart file upload support for sequences
- CI/CD workflows for releases
- Updated audio files including freestyle caller

dpgarceau commented:
I cleaned up a few things from my prior pull request and added the 'wrong pilot' score correction feature into the admin menus

dpgarceau committed:
Remove unused gitignore entries

rolfbartels reviewed (approved):
There was a lot to review here, 
I am approving this pull request, and will test once release.

dpgarceau committed:
Revamp system update feature with version checking and confirmation

- Add fetch_update.sh: lightweight fetcher script for devices
- Update judge_update.sh with distinct exit codes (0=current, 1=error, 2=updated)
- Update judge_setup.sh to install fetch_update.sh instead of full script
- Add /api/system/check-update endpoint for version comparison
- Update /api/system/update to call local script and handle exit codes
- Add confirmation modal with version display and internet warning
- Implement two-phase update flow (check -> confirm -> install)
- Single entry point for both Admin menu and SSH updates

dpgarceau committed:
Add lightweight /api/battery endpoint for fast battery checks

- Add getBatteryPercent() method to InfoCollectorService (I2C only)
- Add /api/battery endpoint returning just {"percent": 85}
- Update checkBatteryForModal() in pilot-list templates to use new endpoint
- ~100x faster than /api/getinfo which runs 4 expensive operations

dpgarceau committed:
Remove all REVIEWED-UNUSED commented imports and fields

Delete 28 commented items across 11 files that were previously
tagged as unused in 2025-11. All verified as truly unused:
- Logger/LoggerFactory in SettingUtils
- JudgeApplication, Properties in CommandLineAppStartupRunner
- ZipEntry in SettingService
- URL, ResourceUtils, Gson in SequenceService
- CompDTO, URL in ScheduleService
- Autowired in RoundsService
- URL in PilotService
- URL, ArrayList, HashMap, List, Map, NodeList in CompService
- ArrayList in FlightsUploadDTO
- ArrayList, List in ScheduleDTO
- Map in RoundsDTO
- ConnectException in GlobalRestExceptionHandler
- SequenceService field in RootController

dpgarceau commented:
@DevDavidW - I have added the changes we spoke about this morning (in three separate commits!) 
- Fixed the system update process so that the front end, ssh, and setup processes all now use a single fetch-update.sh 
- Fixed the battery warning dialogue to use a new get-battery process instead of the slow get-info api
- Removed all the "reviewed-unused" commented out imports

dpgarceau committed:
Fix freestyle mismatch duplicates in score resolve page

DevDavidW commented on judge/src/main/resources/templates/pilot-list-global.html:
```diff
@@ -1,171 +1,298 @@
+        </div>
+
+        <!-- Centered Judge button -->
+        <div style="text-align: center; margin-top: 20px;">
```
In the original code this had a conditional around it to show the judge button only if the pilot was marked as active. Should this have been removed? The "All Rounds Completed" text just below this has the opposite conditional which was retained from the original. Perhaps the pilot still needs to have 0's entered even when the pilot is no longer active?

DevDavidW replied:
Added a commit that filters the pilots for this page so no active pilots will even show. There is no need to navigate past pilots that are marked inactive (either left the contest or never showed up).

DevDavidW commented on TODO.md:
Could we just remove this file in favor of the GitHub project items? That way its self maintained (through commits/PRs completing items) instead of having to remember to also update this file.

DevDavidW commented on docs/v2.0-Release-Notes.md:
The release notes are usually included in the actual release build description. Though it usually follows the commit descriptions, it does allow customized notes. Also, the changelog will already have this same information if we continue to maintain that so I'm not sure we need this extra file?

DevDavidW replied:
Moved these all into the CHANGELOG and the major points into the PR description (to become the release description).

DevDavidW commented on judge/src/main/java/co/za/imac/judge/service/SequenceFolderResolver.java:
```diff
@@ -0,0 +1,339 @@
+
+    private static final Logger logger = LoggerFactory.getLogger(SequenceFolderResolver.class);
+
+    private static final String FIGURES_PATH = SettingUtils.getApplicationConfigPath() + "/figures/en";
```
I think this FIGURES_PATH should use the language setting instead of hard-coding "en" in hopes of this one day supporting multiple languages. But we can add that as an enhancement later.

DevDavidW replied:
I changed this to use the language setting instead

DevDavidW commented on figures/en/default/season.cfg:
Why not let this just be a setting in the settings.json file? That way it can be managed by the settings service remotely (or locally) if needed.

DevDavidW replied:
Moved this into the settings.json file

DevDavidW commented on judge/src/main/java/co/za/imac/judge/service/CompService.java:
```diff
@@ -135,6 +132,28 @@ public boolean saveCompToFile() throws IOException, SAXException, ParserConfigur
+     * Save comp settings locally without any Score server contact.
+     * Updates both in-memory compDTO and the comp.json file.
+     */
+    public boolean saveCompToFileLocal() throws IOException {
```
To avoid duplicate code in the original saveCompToFile() function, should invoke this function from that one. Otherwise, could add a parameter to the original saveCompToFile to bypass Score for that new button press. It would just avoid having the same code in more than one place.

DevDavidW replied:
Fixed this to have one method for the same code.

DevDavidW commented on judge/src/main/resources/templates/newcomp.html:
```diff
@@ -1,287 +1,364 @@
+</head>
+
+<body class="orange lighten-4">
+  <div th:replace="~{banner::line_judge}"></div>
```
Would look better if didn't have the line_judge banner here since this is where it is being set for the first time and the page already will show the same information. This text is close to overwriting the header text anyway.

DevDavidW replied:
Removed the banner

DevDavidW commented on judge/src/main/resources/templates/newcomp.html:
```diff
@@ -1,287 +1,364 @@
+          <div class="setting-label">Scoring Mode:</div>
+          <div class="radio-group">
+            <label class="black-text">
+              <input class="with-gap" name="score_mode" type="radio" value="global" checked />
```
Even though this has a default of "global" being checked, the RootController /newcomp method sets this to a default of byRound which then gets processed on this page which thereby ignores this setting (making this confusing).

DevDavidW replied:
This was resolved by changing the RootController /newcomp method to also default to "global" setting

DevDavidW commented on judge/src/main/resources/templates/newcomp.html:
```diff
@@ -1,287 +1,364 @@
+        dataType: "json",
+        success: function(currentSettings) {
+          // Merge new values
+          currentSettings.line_number = currentLineNumber;
```
These two are incorrect. They are taking the original values when the page are loaded. Instead it needs to get the "current" values from the controls `$('#lineNumber')` and `$('#judgeNumber')` since they could (and will when needed) change.

DevDavidW replied:
This has been fixed in a new commit

DevDavidW commented on judge/src/main/resources/templates/newcomp.html:
```diff
@@ -1,287 +1,364 @@
+                  data: JSON.stringify({
+                    score_mode: selectedMode,
+                    sequences: sequences,
+                    unknown_sequences: 1,
```
The old page also had a selection for the number of unknown sequences. I see here it is defaulted to 1. Is a selection for those no longer needed?

DevDavidW replied:
Confirmed this isn't needed as the button to judge unknowns will be disabled if no unknowns are in the schedule

DevDavidW commented on judge/src/main/java/co/za/imac/judge/service/InfoCollectorService.java:
```diff
@@ -134,8 +136,25 @@ private InfoLine getBatteryStatus() {
+     * Much faster than collectInfo() which runs 4 expensive operations.
+     * @return Battery percentage (0-100), or -1 if read fails
+     */
+    public int getBatteryPercent() {
```
To avoid duplicate code, this function should be called from the one above to get battery status.

DevDavidW replied:
This is now refactored to avoid duplicate code.

DevDavidW commented on judge/src/main/java/co/za/imac/judge/dto/SettingDTO.java:
```diff
@@ -9,30 +9,29 @@ public class SettingDTO {
         this.line_number = line_number;
     }
 
+    public SettingDTO() {
```
I'm not really sure why this is moved down. Typically class constructors, like this one, are found first in the class followed by methods. It was that way before, but now all this did was rearrange existing code.

DevDavidW replied:
This has been fixed

DevDavidW commented on judge/src/main/java/co/za/imac/judge/controller/APIController.java:
```diff
@@ -236,4 +303,495 @@ public InfoJson getLatestInfo() {
+    }
+
+    @PostMapping("/api/sequence/upload")
+    public ResponseEntity<String> uploadSequenceZip(
```
I'm not so sure adding in a method to manage the figures within the judge app is a good idea. The figures management was removed from judge to make it easier to manage them separately - though I know this is for the unknowns. As this is more file management, it could be a similar process to the judge update script where it fetches the file say from the Score api endpoint, and then can unzip the files into the right place. Adding in all this zip management into the judge app is unnecessary as the OS has zip methods by default.

DevDavidW replied:
This has been refactored to pull from Score itself (instead of it being pushed) when doing a sync. Score 4.71 was updated to support this pull method.

DevDavidW committed:
Change to get language from settings for figures path references

DevDavidW committed:
Adjusted pilot service to allow filtering inactive pilots from the carousel's and other needed areas

DevDavidW committed:
Updated pilot list pages which will no longer have inactive pilots

DevDavidW committed:
Fixed admin pages to use proper setting values, adjusted headers to not overwrite banner, fixed updateSettings to work with Windows when developing

DevDavidW committed:
Fixed validation page to show full button name, removed banner as not needed here

DevDavidW committed:
Moved season.year setting into the settings service and related json file

DevDavidW committed:
Refactored getBattery methods to reduce code duplication

DevDavidW committed:
Refactored sameCompToFile to reduce code duplication

DevDavidW committed:
Updated methods to support docker build to move away from older copy plugin which stopped working

DevDavidW committed:
Removed docker "test" files not needed

DevDavidW committed:
Updated setup script to adjust settings.json file as needed

DevDavidW committed:
Increase text size of warning message

DevDavidW committed:
Reduced duplication of short_desc folder checks

DevDavidW committed:
Added option to prompt for year code in new setup script

rolfbartels reviewed (approved):
.

dpgarceau committed:
Refactor sequence_validation handling to consolidate errors and warnings into a simplified structure, enhancing clarity and maintainability. Update related classes and HTML template for improved display of new format.

dpgarceau committed:
Added logic to filter out / ignore  sequence_validation issues for classes without pilots

DevDavidW committed:
Fixed issue with language setting in path for error with image

DevDavidW committed:
Fixed getFiguresPath to fetch language setting instead of using hardcoded value

DevDavidW committed:
Change default scoring mode to global

DevDavidW committed:
Changed to use sh in settingservice to support both device and docker

DevDavidW committed:
Updated startup to output new settings to log and removed unused comments and code

DevDavidW committed:
Added method to retrieve unknown figures zip from Score during new event load and re-sync process

DevDavidW committed:
Changed the upgrade process to only consider newer versions. Helps in case of side-loading a newer version to the devices.

DevDavidW committed:
Consolidated release notes into changelog
